test/test_util.py

```diff
@@ -117,6 +118,14 @@ def test_resolve_dtype_iter_a(self) -> None:
         self.assertEqual(resolve_dtype_iter((a3.dtype, a5.dtype)).kind, 'U')
         self.assertEqual(resolve_dtype_iter((a3.dtype, a5.dtype)).itemsize, 40)
 
+        with pytest.raises(TypeError):
+            self.assertEqual(resolve_dtype_iter((a3.dtype, int)).kind, 'U')
+
+        self.assertEqual(resolve_dtype_iter((a1.dtype,)), a1.dtype)
+
+        with pytest.raises(ValueError):
+            self.assertEqual(resolve_dtype_iter(()), a1.dtype)
```

Comment:
Since these are raising, we don't need the `assertEqual` calls, right?

```suggestion
        with pytest.raises(TypeError):
            resolve_dtype_iter((a3.dtype, int))

        self.assertEqual(resolve_dtype_iter((a1.dtype,)), a1.dtype)

        with pytest.raises(ValueError):
            resolve_dtype_iter(())
```